File: pset6/readability/readability.py
import cs50

text = cs50.get_string("Text: ")

letter_count = 0
word_count = 1
sentence_count = 0

for c in text:
    if c.isalpha():
        letter_count += 1
    if c == ' ':
        word_count += 1
    if c in ['.', '?', '!']:
        sentence_count += 1

L = letter_count / word_count * 100
S = sentence_count / word_count * 100
result = round(0.0588 * L - 0.296 * S - 15.8)

if result < 1:
    print("Before Grade 1")
elif result >= 16:
    print("Grade 16+")
else:
    print(f"Grade {result}")

File: pset2/readability/readability.c
#include<stdio.h>
#include<cs50.h>
#include<string.h>
#include<ctype.h>
#include<math.h>

int main(void)
{
    int letters = 0, words = 0, sentences = 0;
    char *text = get_string("Text: ");
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // count letters
        if (isalpha(text[i]))
        {
            letters++;
        }

        // count words
        if (text[i] == ' ')
        {
            words++;
        }

        // count sentences
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }
    words++; // because we count spaces for words, we lost the last word, therefore we add one more

    // print for debugging
    // printf("Letters: %d\n", letters);
    // printf("Words: %d\n", words);
    // printf("Sentences: %d\n", sentences);

    // calculate L and S and final result
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;
    int result = round(0.0588 * L - 0.296 * S - 15.8);

    // printf("readability: %d\n", result);

    if (result < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (result >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", result);
    }

}
